nifi-extension-bundles/nifi-standard-services/nifi-schema-registry-service-api/src/main/java/org/apache/nifi/schemaregistry/services/SchemaRegistry.java

```diff
@@ -45,4 +45,32 @@ public interface SchemaRegistry extends ControllerService {
      * @return the set of all Schema Fields that are supplied by the RecordSchema that is returned from {@link #retrieveSchema(SchemaIdentifier)}
      */
     Set<SchemaField> getSuppliedSchemaFields();
+
+    /**
+     * Retrieves the raw schema definition including its textual representation and references.
+     * <p>
+     * This method is used to retrieve the complete schema definition structure, including the raw schema text
+     * and any schema references. Unlike {@link #retrieveSchema(SchemaIdentifier)}, which returns a parsed
+     * {@link RecordSchema} ready for immediate use, this method returns a {@link SchemaDefinition} containing
+     * the raw schema content that can be used for custom schema processing, compilation, or when schema
+     * references need to be resolved.
+     * </p>
+     * <p>
+     * This method is particularly useful for:
+     * <ul>
+     *   <li>Processing schemas that reference other schemas (e.g., Protocol Buffers with imports)</li>
+     *   <li>Custom schema compilation workflows where the raw schema text is needed</li>
+     *   <li>Accessing schema metadata and references for advanced schema processing</li>
+     * </ul>
+     * </p>
+     *
+     * @param schemaIdentifier the schema identifier containing id, name, version, and optionally branch information
+     * @return a {@link SchemaDefinition} containing the raw schema text, type, identifier, and references
+     * @throws IOException if unable to communicate with the backing store
+     * @throws SchemaNotFoundException if unable to find the schema based on the given identifier
+     * @throws UnsupportedOperationException if the schema registry implementation does not support raw schema retrieval
+     */
+    default SchemaDefinition retrieveRawSchema(SchemaIdentifier schemaIdentifier) throws IOException, SchemaNotFoundException {
```

Comment:
Recommend renaming this to `retrieveSchemaDefinition()` to align with the returned object type.